lib/derive/src/lib.rs

```diff
@@ -159,43 +159,41 @@ fn derive_struct_fields(data: &DataStruct) -> (TokenStream, TokenStream) {
                                     self.#name.initialize(#name);
                             }
                         }
-                    } else {
-                        if let Some(identifier) = identifier {
-                            let mut access_expr = quote_spanned! {
-                                f.span() =>
-                                    instance.exports.get_with_generics::<#inner_type, _, _>(#identifier)
+                    } else if let Some(identifier) = identifier {
```

Comment:
I don't think this transformation is always a good idea. It's true that they're semantically equivalent at runtime, this transformation is a bit more explicit about whats happening in some ways, and sometimes it does improve code clarity, but sometimes an

```
else { 
  if ... { }
}
```

is forward-thinking in that it indicates that the `else` block is a distinct case that should be handleded as 1 unit. Currently there's only 1 handler in there so it can be simplified to an `else if` but the inverse transformation may be required in the future to make the code clean.

Perhaps that's over-thinking it, but I'm not sure I buy that this transformation is a good idea in the abstract.

-----

Specific to this PR: I think this change is fine, thanks for cleaning up some of the formatting too -- it's too bad there's no way to opt-in to rustfmt formatting macros. Historical note: my primary editor at the time kept choking on this crate (probably due to some combination of the LSP client I was using and the version of rust analyzer) and I had to use a different editor to write large sections of it and the second editor had its own ideas about formatting which is probably why it's so messy here.